### PR TITLE
stylelint, ref #1300

### LINKED ISSUE #1300: Add stylelint
https://github.com/stylelint/stylelint

### PR DESCRIPTION
**Pull request by silentcloud**
close #1300 

First of all, thanks for your contribution! :-)

Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/1318)

### CODE REVIEW

File: components/action-sheet/style/base.less [hunk 1]
```diff
@@ -7,7 +7,7 @@
     bottom: 0;
     left: 0;
     z-index: @action-sheet-zindex;
-    overflow-scrolling: touch;
+    -webkit-overflow-scrolling: touch;
```
Comment (paranoidjk) [inline components/action-sheet/style/base.less]: is this  stylint requirement? why ? I think autoPrefixer will do this automatic

File: components/action-sheet/style/index.less [hunk 2]
```diff
@@ -10,8 +10,9 @@
   bottom: 0;
   width: 100%;
   background-color: @fill-base;
+
   &&-share {
-    background-color: #F2F2F2;
+    background-color: #f2f2f2;
```
Comment (paranoidjk) [inline components/action-sheet/style/index.less]: IMO, normal css coding style seems to prefer use uppercase?

File: components/button/style/index.less [hunk 3]
```diff
@@ -55,8 +56,8 @@
     font-size: @button-font-size-sm;
     height: @button-height-sm;
     line-height: @button-height-sm;
-    .border-radius(@radius-md);
     padding: 0 @h-spacing-lg;
+    .border-radius(@radius-md);
```
Comment (paranoidjk) [inline components/button/style/index.less]: less function must be placed at bottom?

File: components/drawer/style/index.less [hunk 4]
```diff
@@ -13,15 +13,17 @@
   &-sidebar {
     z-index: 2;
     position: absolute;
-    transition: transform .3s ease-out;
+    transition: transform 0.3s ease-out;
```
Comment (paranoidjk) [inline components/drawer/style/index.less]: still weird, I remember normal css coding style says that do not use prefix 0 is better...

why stylelint rules looks like different  with our habit

File: components/list/style/index.less [hunk 5]
```diff
@@ -46,55 +49,65 @@
   // box-sizing: content-box;
   vertical-align: middle;
   overflow: hidden;
-  .box-align;
   transition: background-color 200ms;
+  .box-align;
```
Comment (paranoidjk) [inline components/list/style/index.less]: not **.box-align()** ? whats this syntax ?

File: components/slider/style/index.less [hunk 6]
```diff
@@ -46,7 +46,6 @@
 
     &:focus {
       background-color: tint(@brand-primary, 20%);
-      background: #000;
```
Comment (paranoidjk) [inline components/slider/style/index.less]: 这个是不是降级的？

File: components/action-sheet/style/base.less (hunk 1 above)
Comment (silentcloud) [inline components/action-sheet/style/base.less]: @paranoidjk stylint 报错，overflow-scrolling 不标准的 css

File: components/action-sheet/style/index.less (hunk 2 above)
Comment (silentcloud) [inline components/action-sheet/style/index.less]: @paranoidjk 同样，stylelint 小写

File: components/button/style/index.less (hunk 3 above)
Comment (silentcloud) [inline components/button/style/index.less]: 不是说要在 bottom，如果放在上面需要在 .border-radius() 下面多个空行出来，看着奇怪

File: components/list/style/index.less (hunk 5 above)
Comment (silentcloud) [inline components/list/style/index.less]: 原来是这么写的，我只是移位，不晓得原来的情况

File: components/slider/style/index.less (hunk 6 above)
Comment (silentcloud) [inline components/slider/style/index.less]: 不清楚，感觉冲突，stylelint 也在报错

File: components/action-sheet/style/base.less (hunk 1 above)
Comment (paranoidjk) [inline components/action-sheet/style/base.less]: 这里可能要去stykelint那边搜索下，我们是依赖autoPrefixer来加前缀的

Comment (silentcloud) [inline components/action-sheet/style/base.less]: 那我改一下规则好了

File: components/list/style/index.less (hunk 5 above)
Comment (silentcloud) [inline components/list/style/index.less]: @paranoidjk 这里没问题的，定义里写了默认参数，使用的时候可以不传参， ref http://lesscss.org/features/#mixins-parametric-feature

File: components/drawer/style/index.less (hunk 4 above)
Comment (silentcloud) [inline components/drawer/style/index.less]: 恩 google styleguide 里确实是建议不加 0 的

File: components/action-sheet/style/index.less (hunk 2 above)
Comment (paranoidjk) [inline components/action-sheet/style/index.less]: 这个勉强能接受吧，不过也还是跟我以前学的 css style guide 不符合，以前都是说要颜色值大写...

File: components/list/style/index.less (hunk 5 above)
Comment (paranoidjk) [inline components/list/style/index.less]: ok, get 了

File: components/action-sheet/style/base.less (hunk 1 above)
Comment (paranoidjk) [inline components/action-sheet/style/base.less]: 杯具。。这还是个bug ，-webkit-overflow-scrolling 是非标准属性，autoPrefixer不会处理的，必须我们自己加webkit前缀。改回去吧，顺便check下有没有其他的
http://npmcdn.alibaba-inc.com/antd-mobile@1.1.3/dist/antd-mobile.css

File: components/slider/style/index.less (hunk 6 above)
Comment (paranoidjk) [inline components/slider/style/index.less]: 这又是个bug，你这样改是对的

```css
    background-color: #40a5ed;
    background: #000; // 优先级更高，但是其实是要上一条规则
```

File: components/action-sheet/style/base.less (hunk 1 above)
Comment (afc163) [inline components/action-sheet/style/base.less]: 用 `/* stylelint-disable-line block-no-empty */` 这种注释掉。

**Pull request merged**